clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -542,8 +544,16 @@ static Map<String, Object> appendSerializerToConfig(Map<String, Object> configs,
         Map<String, Object> newConfigs = new HashMap<>(configs);
         if (keySerializer != null)
             newConfigs.put(KEY_SERIALIZER_CLASS_CONFIG, keySerializer.getClass());
+        else {
+            if (newConfigs.get(KEY_SERIALIZER_CLASS_CONFIG) == null)
```

Comment:
Similar to what Chris Egerton has suggested above these can also be collapsed into `else-if`

Reply:
thanks